src/uu/tail/src/args.rs

```diff
@@ -358,8 +421,35 @@ fn parse_num(src: &str) -> Result<Signum, ParseSizeError> {
 }
 
 pub fn parse_args(args: impl uucore::Args) -> UResult<Settings> {
-    let matches = uu_app().try_get_matches_from(arg_iterate(args)?)?;
-    Settings::from(&matches)
+    let args_vec: Vec<OsString> = args.collect();
+    let clap_result = match uu_app().try_get_matches_from(args_vec.clone()) {
+        Ok(matches) => {
+            let settings = Settings::from(&matches)?;
+            if !settings.is_default() {
+                // non-default settings can't have obsolete arguments
+                return Ok(settings);
+            }
```

Comment:
Why can default settings have obsolete arguments? Seems to me that if clap succeeds we should always use that value? What is a case where that's not true?

Reply:
> ```shell
> tail -f -f
> ```
> 
> Should try to read the file called `-f`. 

> 
> ```shell
> tail -f -f -f
> ```
> 
> should be allowed. 

No, in GNU's implementation both activate -f and use STDIN as input.

> If we fix that by adding `.overrides_with(option::FOLLOW)` to clap. 

IMHO, we should allow overrides, as this is what GNU does. `tail (-f)+` is basically the same `tail -f --`, no matter how much `-f` follow. 

> Then `-f -f` is not parsed as obsolete syntax in your implementation.

Obsolete should only be parsed for following format: `tail <args> <file>?` (where args can be `-f+`).
so `tail -f -f -f` would basically be `tail -fff` in obsolete syntax, semantically resulting to `tail -f --`.

Reply:
AH, wrong window...

Answer to this: Problem is e.g. `tail +c`. This is handled as valid by clap (default tail to file `+c`).

Reply:
Ohhh you're totally right! Sorry!